cypress/e2e/signUp.cy.js

```diff
@@ -1,12 +1,39 @@
 /// <reference types="cypress" />
 /// <reference types="../support" />
 
+import SignUpPageObject from '../support/pages/signUp.pageObject';
+import HomePageObject from '../support/pages/home.pageObject';
+
+const signUpPage = new SignUpPageObject();
+const homePage = new HomePageObject();
+
 describe('Sign Up page', () => {
-  before(() => {
+  let user;
+
+  beforeEach(() => {
+    cy.task('db:clear');
+    cy.task('generateUser').then(generateUser => {
+      user = generateUser;
```

Comment:
The user generation task should be encapsulated inside a helper method. This will reduce the complexity of the test setup and make the code cleaner.